src/main/java/com/linkedin/kmf/services/ConsumeService.java

```diff
@@ -153,10 +153,10 @@ private void consume() throws Exception {
         _sensors._consumeError.record();
         continue;
       }
-      int partition = record.partition();
-      long index = (Long) avroRecord.get(DefaultTopicSchema.INDEX_FIELD.name());
+      int partition = (int) avroRecord.get(DefaultTopicSchema.PARTITION_FIELD.name());
```

Comment:
So @WarrenGreen  is concerned with the source message ending up in the wrong destination partition after mirroring since MM will do round robin when the source message does not have a key. Yes?  Without a key the consumer on the destination topic might see out-of-order messages with respect to the original (topic, partition) in which they were produced.  Is this the problem?